File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/LinkedStory.java
package quintype.com.templatecollectionwithrx.models.story;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

/**
 * Created by rakshith on 9/26/17.
 */
public class LinkedStory implements Parcelable {
    @SerializedName("headline")
    private
    String mHeadline;
    @SerializedName("story-content-id")
    private
    String storyContentId;
    @SerializedName("id")
    private
    String mStoryId;

    @Override
    public int describeContents() {
        return 0;
    }

    public String getmHeadline() {
        return mHeadline;
    }

    public void setmHeadline(String mHeadline) {
        this.mHeadline = mHeadline;
    }

    public String getStoryContentId() {
        return storyContentId;
    }

    public void setStoryContentId(String storyContentId) {
        this.storyContentId = storyContentId;
    }

    public String getmStoryId() {
        return mStoryId;
    }

    public void setmStoryId(String mStoryId) {
        this.mStoryId = mStoryId;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(this.mHeadline);
        parcel.writeString(this.storyContentId);
        parcel.writeString(this.mStoryId);
    }

    public LinkedStory(Parcel in) {
        this.mHeadline = in.readString();
        this.storyContentId = in.readString();
        this.mStoryId = in.readString();
    }

    @Override
    public String toString() {
        return "LinkedStory{" +
                "mHeadline='" + mHeadline + '\'' +
                ", storyContentId='" + storyContentId + '\'' +
                ", mStoryId='" + mStoryId + '\'' +
                '}';
    }

    public static final Creator<LinkedStory> CREATOR = new Creator<LinkedStory>() {
        @Override
        public LinkedStory createFromParcel(Parcel in) {
            return new LinkedStory(in);
        }

        @Override
        public LinkedStory[] newArray(int size) {
            return new LinkedStory[size];
        }
    };

    public static Creator<LinkedStory> getCREATOR() {
        return CREATOR;
    }

}

File: androidcore/src/main/java/com/example/androidcore/models/story/LinkedStory.kt
package com.example.androidcore.models.story

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

/**
 * Created TemplateCollectionWithRx by rakshith on 9/4/18.
 */

class LinkedStory(parcel: Parcel) : Parcelable {
    @SerializedName("headline")
    private var mHeadline: String? = null
    @SerializedName("story-content-id")
    var storyContentId: String? = null
    @SerializedName("id")
    private var mStoryId: String? = null

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(this.mHeadline)
        parcel.writeString(this.storyContentId)
        parcel.writeString(this.mStoryId)
    }

    init {
        this.mHeadline = parcel.readString()
        this.storyContentId = parcel.readString()
        this.mStoryId = parcel.readString()
    }

    override fun toString(): String {
        return "LinkedStory{" +
                "mHeadline='" + mHeadline + '\''.toString() +
                ", storyContentId='" + storyContentId + '\''.toString() +
                ", mStoryId='" + mStoryId + '\''.toString() +
                '}'.toString()
    }

    companion object CREATOR : Parcelable.Creator<LinkedStory> {
        override fun createFromParcel(parcel: Parcel): LinkedStory {
            return LinkedStory(parcel)
        }

        override fun newArray(size: Int): Array<LinkedStory?> {
            return arrayOfNulls(size)
        }
    }
}